src/libaktualizr/uptane/uptane_test.cc

```diff
@@ -295,6 +295,27 @@ TEST(Uptane, UptaneSecondaryAdd) {
   EXPECT_TRUE(ecu_data["ecus"][1]["clientKey"]["keyval"]["public"].asString().size() > 0);
 }
 
+/* Adding multiple secondaries with the same serial throws an error */
+TEST(Uptane, UptaneSecondaryAddSameSerial) {
+  TemporaryDirectory temp_dir;
+  auto http = std::make_shared<HttpFake>(temp_dir.Path());
+  Config config;
+  boost::filesystem::copy_file("tests/test_data/cred.zip", temp_dir / "cred.zip");
+  config.provision.provision_path = temp_dir / "cred.zip";
+  config.provision.mode = ProvisionMode::kAutomatic;
+  config.uptane.director_server = http->tls_server + "/director";
+  config.uptane.repo_server = http->tls_server + "/repo";
+  config.tls.server = http->tls_server;
+  config.provision.primary_ecu_serial = "testecuserial";
+  config.storage.path = temp_dir.Path();
+  config.pacman.type = PackageManager::kNone;
+  UptaneTestCommon::addDefaultSecondary(config, temp_dir, "secondary_ecu_serial", "secondary_hardware");
+  UptaneTestCommon::addDefaultSecondary(config, temp_dir, "secondary_ecu_serial", "secondary_hardware");
+
+  auto storage = INvStorage::newStorage(config.storage);
+  EXPECT_THROW(SotaUptaneClient::newTestClient(config, storage, http), std::runtime_error);
+}
+
```

Comment:
Looks good. I have a few suggestions for this one:
1. I think we can remove most of the configuration. Most of it will never get touched because we fail so early.
2. Change the hardware ID of the second secondary.
3. After the failure, change the serial of the second secondary and try again to prove that it should work.

Reply:
> 3. After the failure, change the serial of the second secondary and try again to prove that it should work.
We have had bug here. We didn't clear misconfigured ecus.

Reply:
> We didn't clear misconfigured ecus.

That is true and that's fine for now. But you can clear the misconfiguration table manually, right?

Reply:
I have fixed it, then commeneted here, but haven't pushed changes.

Reply:
I think points 1 and 3 are covered now, but I think point 2 still stands: I recommend changing the hardware ID of the second secondary.